services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -80,6 +82,29 @@ def check_indexable(feature: FeatureType) -> None:
     return indexable_columns
 
 
+def get_delete_operations(
+    all_repo_files: set[str], split_names: set[str], config: str, index_file_location: str
+) -> list[CommitOperationDelete]:
+    only_config_files: set[str] = {
+        file for file in all_repo_files if re.compile(f"^({re.escape(config)})/").match(file)
+    }
+    pattern_only_config_splits_files = re.compile(
+        f"^({'|'.join(re.escape(f'{config}/{split_name}') for split_name in split_names)})/"
+    )
+
+    # delete old files from non existent splits
+    files_to_ignore: set[str] = {
+        file
+        for file in only_config_files
+        # ignore existing files in all current splits
+        if pattern_only_config_splits_files.match(file)
+        # but not the old index in split
+        and file != index_file_location
+    }
+    files_to_delete = only_config_files - files_to_ignore
```

Comment:
to simplify the logic and avoid double negations, maybe set files_to_delete with the loop?

Something like
```suggestion
    files_to_delete: set[str] = {
        file
        for file in only_config_files
        # delete the obsolete index in the current split
        if file == index_file_location
        # and files from unknown splits
        or not pattern_only_config_splits_files.match(file)
    }
```